viadot/flows/transform_and_catalog.py

```diff
@@ -0,0 +1,245 @@
+import os
+from pathlib import Path
+import shutil
+from typing import Dict, List, Union
+
+from prefect import Flow, task
+from prefect.tasks.shell import ShellTask
+
+from viadot.tasks import CloneRepo, AzureKeyVaultSecret, LumaIngest
+
+
+@task
+def _cleanup_repo(dbt_repo_dir_name: str) -> None:
+    """
+    Remove a repo folder.
+
+    Args:
+        dbt_repo_dir_name (str): The name of the temporary folder.
+    """
+    shutil.rmtree(dbt_repo_dir_name, ignore_errors=True)  # Delete folder on run
+
+
+class TransformAndCatalog(Flow):
+    """Build specified dbt model(s) and upload the generated metadata to DataHub or other destination."""
+
+    def __init__(
+        self,
+        name: str,
+        dbt_project_path: str,
+        dbt_repo_url: str = None,
+        dbt_repo_url_secret: str = None,
+        dbt_repo_branch: str = None,
+        token: str = None,
+        token_secret: str = None,
+        local_dbt_repo_path: str = None,
+        dbt_selects: Dict[str, str] = None,
+        dbt_target: str = None,
+        stateful: bool = False,
+        metadata_dir_path: Union[str, Path] = None,
+        luma_endpoint: str = "http://localhost/api/v1/dbt",
```

Comment:
It's `luma_url` now (--luma-url), see https://github.com/dyvenia/lumaCLI#luma-dbt

Reply:
Done -> 0cfa9e38